cli/checkpoint/lua/play.lua

```diff
@@ -0,0 +1,142 @@
+-- This is a script that play the contents of .snap/.xlog files to another Tarantool instance.
+-- The files and uri passes throught "TT_CLI_PLAY_FILES_AND_URI".
+-- The --show-system flag passes throught "TT_CLI_PLAY_SHOW_SYS".
+-- The --space flags passes throught "TT_CLI_PLAY_SPACES".
+-- The --from flag passes throught "TT_CLI_PLAY_FROM".
+-- The --to flag passes throught "TT_CLI_PLAY_TO".
+-- The --replica flags passes throught "TT_CLI_PLAY_REPLICAS".
+
+local log = require('log')
+local xlog = require('xlog')
+local json = require('json')
+local netbox = require('net.box')
+
+local function find_in_list(id, list)
+    if type(list) == 'number' then
+        return id == list
+    end
+    for _, val in ipairs(list) do
+        if val == id then
+            return true
+        end
+    end
+    return false
+end
+
+local function filter_xlog(gen, param, state, opts, cb)
+    local from, to, spaces = opts.from, opts.to, opts.space
+    local show_system, replicas = opts['show-system'], opts.replica
+
+    for lsn, record in gen, param, state do
+        local sid = record.BODY and record.BODY.space_id
+        local rid = record.HEADER.replica_id
+        if replicas and #replicas == 1 and replicas[1] == rid and lsn >= to then
+            -- Stop, as we've finished reading tuple with lsn == to
+            -- and the next lsn's will be bigger.
+            break
+        elseif (lsn < from) or (lsn >= to) or
+           (not spaces and sid and sid < 512 and not show_system) or
+           (spaces and (sid == nil or not find_in_list(sid, spaces))) or
+           (replicas and not find_in_list(rid, replicas)) then
+            -- Pass this tuple, luacheck: ignore.
+        else
+            cb(record)
+        end
+    end
+end
+
+local function play(positional_arguments, keyword_arguments)
+    local opts = keyword_arguments
+    local uri = table.remove(positional_arguments, 1)
+    if uri == nil then
+        log.error("Internal error: empty URI is provided")
+        os.exit(1)
+    end
+    local remote = netbox.new(uri)
+    if not remote:wait_connected() then
+        log.error("Fatal error: no connection to the host '%s'", uri)
+        os.exit(1)
+    end
+    for _, file in ipairs(positional_arguments) do
+        print(string.format("• Play is processing file '%s' •", file))
```

Comment:
AFAIU the result will be displayed only at the end of work. If so, why do we need to print the current status of work?

Reply:
> AFAIU the result will be displayed only at the end of work.

The result will really be output only after the completion of the command. 

> why do we need to print the current status of work?

I suppose this is necessary for better readability when an error occurs somewhere in the middle of the list of files that being transferred. We see pairs:

```
(Example)
• Play is processing file AAA •
• Done with file AAA •
• Play is processing file BBB •
• Done with file BBB •
• Play is processing file CCC •
   Result of play: ER_TUPLE_FOUND: Duplicate key exists ...
```